### PR TITLE
Safe refresh

### PR DESCRIPTION
**Pull request by brad**
@orcasgit/orcas-developers This has turned out to be a bit of a refactor, but I think refreshing tokens works better now. The changes:
- New `expires_at` and `timezone` fields
- More careful updating of saved tokens when refreshed. We check that the new expires_at is greater than the old and that the expires_at time is greater than now.
- Added the `transaction.atomic()` decorator to the `get_fitbit_data` method.
- more/better error handling for celery tasks. I think it's set up now so that every call to the API handles rate limit exceptions gracefully
- Use timezone from the `UserFitbit` object rather than dealing with the session
- PEP8 updates

### CODE REVIEW

File: fitapp/migrations/0006_auto_20160421_1204.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+            field=models.FloatField(default=0),
+            preserve_default=False,
+        ),
+        migrations.AddField(
+            model_name='userfitbit',
+            name='timezone',
+            field=models.CharField(default='UTC', max_length=128),
+            preserve_default=False,
+        ),
+    ]
```
Comment (percyperez) [inline fitapp/migrations/0006_auto_20160421_1204.py]: @brad What do you think about renaming this file to `0006_add_expires_timezone_fields.py`?

Comment (brad) [inline fitapp/migrations/0006_auto_20160421_1204.py]: @percyperez Oh yeah, of course!

File: fitapp/models.py [hunk 2]
```diff
@@ -13,6 +13,9 @@ class UserFitbit(models.Model):
     access_token = models.TextField()
     auth_secret = models.TextField()
     refresh_token = models.TextField()
+    # We will store the timestamp float number as it comes from Fitbit here
+    expires_at = models.FloatField()
+    timezone = models.CharField(max_length=128)
```
Comment (percyperez) [inline fitapp/models.py]: @brad I see that the migration file for the `timezone` field has `default='UTC'`. Did you remove `default='UTC'` here after the migration file was created? If not, how `default='UTC'` was added to the migration file?

Comment (brad) [inline fitapp/models.py]: @percyperez When I ran `makemigrations` it asked for a default and I specifie `UTC`

Comment (percyperez) [inline fitapp/models.py]: @brad Ah Cool 👍

File: fitapp/tasks.py [hunk 3]
```diff
@@ -98,6 +112,22 @@ def get_time_series_data(fitbit_user, cat, resource, date=None):
...
             raise Reject(exc, requeue=False)
     except Exception:
-        exc = sys.exc_info()[1]
-        logger.exception("Exception updating data: %s" % exc)
-        raise Reject(exc, requeue=False)
+        _generic_task_exception(sys.exc_info()[1], 'get_time_series_data')
+
+
+@shared_task
+def update_user_timezone(fitbit_user):
```
Comment (percyperez) [inline fitapp/tasks.py]: 👍

**Comment by brad** [comment]
Superseded by #33

**Pull request closed without merging**